### ISSUE TITLE
Accessibility: Screen readers don't have context about pagination

### ISSUE THREAD

**Issue by GuySartorelli** [silverstripe/silverstripe-admin#1969]
WCAG 2.1 1.3.1 4.1.2

Pagination "doesn't read as pagination" according to an accessibility review. There's no context telling a screen reader that the pagination is, in fact, pagination.
https://design-system.w3.org/components/pagination.html gives a good example of accessible pagination - there are boundless other examples online as well.

Main takeaways I can see:
- use the `nav` element as a parent of both a heading (if appropriate) and the pagination list (alternatively use `role="navigation"` which won't affect selectors)
- add either a heading inside the `nav` or a `aria-label` attribute to the `nav` element that has a visually-hidden text "pagination"
- "The current page is indicated by `aria-current="page"`"
- "Where there are multiple `<nav>` elements on a single page, they should all have a unique accessible name." I presume this also applies to elements with `role="navigation"`

## Notes

- The pagination in gridfields and ~campaign admin~ (Steve: I don't think campaign-admin paginates see my comment below) were explicitly flagged, but check _all_ pagination, e.g. in asset admin

## Acceptable criteria

- [ ] Pagination in the CMS is appropriately accessible to screen readers
- [ ] The changelog clearly calls out any changes to markup selectors so people can adjust their custom css/js as necessary

## Kitchen sink CI
- https://github.com/creative-commoners/recipe-kitchen-sink/actions/runs/19217051568

## PRs
- https://github.com/silverstripe/silverstripe-framework/pull/11903
- https://github.com/silverstripe/silverstripe-admin/pull/2053
- https://github.com/silverstripe/silverstripe-versioned-admin/pull/434
- https://github.com/silverstripe/silverstripe-asset-admin/pull/1634
- https://github.com/silverstripe/developer-docs/pull/842

**Comment by emteknetnz** [comment]
I don't think that campaign admin has pagination.

Here's a test script to create 200 campaigns, they all just show without paginating

```php
// create 200 test campaigns to show they do not paginate
use SilverStripe\Versioned\ChangeSet;
$existingCount = ChangeSet::get()->count();
if ($existingCount < 200) {
    for ($i = $existingCount + 1; $i <= 200; $i++) {
        $campaign = ChangeSet::create([
            'Name' => 'Test Campaign ' . $i,
            'Description' => 'Test campaign generated for pagination testing #' . $i,
            'State' => $i % 3 === 0 ? ChangeSet::STATE_PUBLISHED : ChangeSet::STATE_OPEN,
        ]);
        $campaign->write();
    }
}
die;
```

**Comment by GuySartorelli** [comment]
PRs merged
